### PR TITLE
`print_review.py` refactoring

### PR DESCRIPTION
**Pull request by GirZ0n**
**Changes**:
1) Renamed `ReviewResult` -> `GeneralReviewResult`.
2) Added `ReviewResult` from which `GeneralReviewResult` and `FileReviewResult` are now inherited.
3) Put common code when printing `GeneralReviewResult` in different formats in the `convert_review_result_to_json_dict` function.
4) Also prepared the code for further addition of difficulty levels.
5) Fixed the test: now the json schema is checked more strictly. Also I merged `test_multi_file_project.py` and `test_single_file_json_format.py` into one file `test_json_format.py`.
6) Added new tests.

### CODE REVIEW

File: src/python/review/reviewers/review_result.py [hunk 1]
```diff
@@ -8,23 +8,17 @@
...
 
-    @property
-    def all_issues(self) -> List[BaseIssue]:
-        issues = []
-        for file_review_result in self.file_review_results:
-            issues.extend(file_review_result.issues)
 
-        return issues
+@dataclass
+class GeneralReviewResult(ReviewResult):
```
Comment (nbirillo) [inline src/python/review/reviewers/review_result.py:23]: Can we add small comments with descriptions for different ReviewResult classes?

**Review by nbirillo — CHANGES_REQUESTED** [review]
Great! I have a question:
- Why you deleted test_multi_file_project.py and test_single_file_json_format.py? These tests are connected to different cases (one file and several files). We need these tests

**Comment by GirZ0n** [comment]
@nbirillo I merged `test_multi_file_project.py` and `test_single_file_json_format.py` into one file `test_json_format.py` :slightly_smiling_face:

**Comment by nbirillo** [comment]
Sorry, I missed it

**Review by nbirillo — APPROVED** [review]
Great, Thank you!

**Pull request merged**